### ISSUE TITLE
Fwupd can use a bump from 1.7.2 to 1.7.3 to support StarBook Mk V

### ISSUE THREAD

**Issue by kciredor** [NixOS/nixpkgs#153238]
###### Checklist

- [x] Checked the [nixpkgs master branch](https://github.com/NixOS/nixpkgs)

- [x] Checked the [nixpkgs pull requests](https://github.com/NixOS/nixpkgs/pulls)

###### Project name
`nix search` name: fwupd (unsure if dependency fwupd-efi needs a bump as well)

current version: 1.7.2
desired version: 1.7.3

###### Notify maintainers

maintainers: @jtojnar

###### Note for maintainers

Please tag this issue in your PR.

Thanks for (hopefully) picking this up. The latest minor contains Star Labs patches allowing for fwupd to update Coreboot firmware on StarBook Mk V laptops.

**Comment by danth** [comment]
Star Labs firmware updates also require the `flashrom` plugin to be enabled:

https://github.com/NixOS/nixpkgs/blob/a2a0d71f35bae0e8675d602115cdcaeb9ea621cd/pkgs/os-specific/linux/firmware/fwupd/default.nix#L72-L75

I have successfully built fwupd locally, with the plugin, and using Star Labs' fork of both [flashrom](https://github.com/StarLabsLtd/flashrom) and [fwupd](https://github.com/StarLabsLtd/fwupd), although not on the most recent commit. This was able to update Coreboot successfully. The embedded controller update failed and led to NixOS being unable to boot (more information in [this comment](https://github.com/StarLabsLtd/firmware/issues/24#issuecomment-1007948816)).

**Comment by danth** [comment]
Fwupd is now updated past the required version. We just need to get `flashrom` enabled.

**Comment by R-VdP** [comment]
@danth do you still have the patch lying around to enable `flashrom` in `fwupd`? I tried with an overlay setting `haveFlashrom = true` but it seems meson cannot find `flashrom` for some reason. Did you run into the same issue?

**Comment by jtojnar** [comment]
`flashrom` is missing a pkg-config file. That file will be generated when we switch flashrom to Meson build system (add `meson` and `ninja` to `nativeBuildInputs`). It will be necessary to look into the source what is the replacement for `CONFIG_JLINK_SPI` https://cs.github.com/flashrom/flashrom?q=CONFIG_JLINK_SPI.

**Comment by R-VdP** [comment]
@jtojnar thanks for the pointers! I got it working locally now. I think it would be useful to make a PR to get this in nixpkgs, no? And I'd like to change fwupd also so that we can at least enable flashrom easily with an override, because now it's kind of annoying that the variable controlling whether we build with flashrom support, is not overridable.

**Comment by jtojnar** [comment]
@R-VdP Yes, PR would be appreciated.

**Comment by danth** [comment]
Yeah, to get it to be found, I generated the pkg-config file via a script in `fixupPhase` of the `flashrom` package.

Converting to Meson seems like a better solution than that, though.

+1 for a PR

**Comment by R-VdP** [comment]
I opened https://github.com/NixOS/nixpkgs/pull/212818 for this.

**Comment by Roger** [comment]
I found this issue while trying to use fwupd in my StarBook Mk VI. Since the PR to use meson in flashrom was closed because meson is not yet stable, even in latest version. I ended up copying and patching flashrom package [here](https://github.com/Roger/flashrom-meson-nix/) and using it as an overlay, with that you can enable flashrom like this:

```nix
  services.fwupd = {
    enable = true;
    package = pkgs.fwupd.override {
      enableFlashrom = true;
    };
  };
```

And with that, I was able to update coreboot.
